packages/shared/scope-hoisting/src/concat.js

```diff
@@ -139,7 +145,7 @@ async function processAsset(bundle: Bundle, asset: Asset) {
   let statements: Array<Statement>;
   if (asset.astGenerator && asset.astGenerator.type === 'babel') {
     let ast = await asset.getAST();
-    statements = nullthrows(ast).program.program.body;
+    statements = t.cloneNode(nullthrows(ast).program.program).body;
```

Comment:
@mischnic do you know the perf impact of this in the editor benchmark?

Reply:
if Babel AST don't have cycles and are simple serializable structures, maybe `JSON.parse(JSON.stringify(nullthrows(ast).program.program))` is faster?